pyk/src/pyk/kcfg.py

```diff
@@ -50,6 +51,11 @@ def id(self) -> str:
         def to_dict(self) -> Dict[str, Any]:
             return {'id': self.id, 'term': self.cterm.term.to_dict()}
 
+        def __lt__(self, other):
+            if not isinstance(other, KCFG.Node):
+                return NotImplemented
+            return self.id < other.id
+
```

Comment:
Can't this be enforced with our type-checker?

```suggestion
        def __lt__(self, other: KCFG.Node) -> bool:
            return self.id < other.id

```

Reply:
That's a `return`, not `raises`. I think python will call the other classes implementation to see if that works.